src/storage/src/hummock/sstable/block.rs

```diff
@@ -310,13 +328,23 @@ impl Block {
         self.restart_points.partition_point(pred)
     }
 
-    pub fn data(&self) -> &[u8] {
-        &self.data[..self.data_len]
+    pub fn slice(&self, range: impl RangeBounds<usize>) -> &[u8] {
+        let start = range.start().unwrap_or_default();
+        let end = range.end().unwrap_or(self.data_len);
+        &self.data[start..end]
```

Comment:
Shall we forward the idx range to slice directly just like how we originally use the range?

Reply:
I didn't found any function of slice that receives a `impl RangeBounds<usize>` to forward it directly. I'll restore the function to `fn data(&self) -> &[u8]` as it does include any other operations any more.